### PR TITLE
app-services tabs component constructor now wants a database filename

### PR DESCRIPTION
**Pull request by mhammond**
This is the matching PR for https://github.com/mozilla/application-services/pull/4862. Marking as draft because there's no a-s release yet, but CC @grigoryk @jonalmeida

### CODE REVIEW

File: components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/RemoteTabsStorage.kt [hunk 1]
```diff
@@ -18,13 +19,16 @@ import mozilla.components.support.utils.logElapsedTime
 import mozilla.appservices.remotetabs.InternalException as RemoteTabProviderException
 import mozilla.appservices.remotetabs.TabsStore as RemoteTabsProvider
 
+private const val TABS_DB_NAME = "tabs.sqlite"
+
 /**
  * An interface which defines read/write methods for remote tabs data.
  */
 open class RemoteTabsStorage(
+    private val storageDir: File,
```
Comment (jonalmeida) [inline components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/RemoteTabsStorage.kt:28]: We should add a changelog entry about this breaking API before landing.

File: components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/storage/SyncedTabsStorage.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
 class SyncedTabsStorage(
     private val accountManager: FxaAccountManager,
     private val store: BrowserStore,
-    private val tabsStorage: RemoteTabsStorage = RemoteTabsStorage()
+    private val tabsStorage: RemoteTabsStorage,
```
Comment (jonalmeida) [inline components/feature/syncedtabs/src/main/java/mozilla/components/feature/syncedtabs/storage/SyncedTabsStorage.kt:30]: Same for here too.

**Review by jonalmeida — COMMENTED** [review]
LGTM, just needs a changelog entry as well when we're ready to land this with the A-S upgrade.

**Comment by mhammond** [comment]
There are v92.0.* versions of app-services now available, so I guess this is ready to roll. I'm not sure how to make CI happy about this though unless I attempt to bump a-s in this PR, but I don't think that's the correct thing to do. cc @jhugman who is presumably also trying to get nimbus upgraded.

**Comment by bendk** [comment]
Can we get an r+ for this one?  I'm hoping to get a new app-services release out and we need this to to resolve the breaking changes.  I wouldn't merge the code, just looking for approval.

**Review by Mugurell — APPROVED** [review]
Rubberstamping this since it looked good to Jonathan and the changelog entries were added.

This will need to be rebased to at least update the changelog but in such probably the CI will get happy too.

**Comment by bendk** [comment]
I just made #12044 which includes this commit and also bumps the app-services version to `93.0.1`.  I think we can close this one.

**Pull request closed without merging**